### PR TITLE
Improve Pcap & PcapNg support (particularly timestamps)

### LINKED ISSUE #195: Nanosecond libpcap
Hi,

I wish I could use scapy for nanosecond analysis, but I get a 
"Not a supported capture file", when trying to read a pcap encoded as Nanosecond lib pcap.
(I'm using the latest revision)

Enclosed the same 3 packets encoded by wireshark with different settings.
the nanosecondlibpcap.pcap file is the one I'd like to be able to read.
We should be able to see : 
packet 1 at 15:21:09:000150304
packet 2 at 15:21:09:000169406
packet 3 at 15:21:09:000216777

Although pcapng seems to be able to use nanoseconds, somehow the nanosecond part is lost (at least when visualizing with wireshark)

[pcaps.zip](https://github.com/secdev/scapy/files/311692/pcaps.zip)

### PR DESCRIPTION
**Pull request by p-l-**
This fixes #195 (support for Pcap files with nanosecond precision timestamp) + another (unreported) issue: timestamps are wrong in PcapNg format, discovered when writing new tests.

This also add support for arbitrary time precision in PcapNg, and for two (less common but sometimes found) more packet blocks.

New tests have been added to prevent regressions.

### CODE REVIEW

File: scapy/utils.py [hunk 1]
```diff
@@ -824,14 +832,15 @@ def read_packet(self, size=MTU):
 
     def read_block_idb(self, block, _):
         """Interface Description Block"""
+        # We should read options to set if_tsresol
```
Comment (gpotter2) [inline scapy/utils.py]: I don't really understand what "We should" means...

From what i understand, it's a feature that needs to be added in the future, right ? If so, it gonna be lost forever in that piece of code...

File: scapy/utils.py [hunk 2]
```diff
@@ -854,7 +863,9 @@ def read_packet(self, size=MTU):
             if conf.debug_dissector:
                 raise
             p = conf.raw_layer(s)
-        p.time = sec+0.000001*usec
+        # We should use if_tsresol when available (see
```
Comment (gpotter2) [inline scapy/utils.py]: Same

File: scapy/utils.py [hunk 3]
```diff
@@ -806,15 +816,18 @@ def read_packet(self, size=MTU):
             try:
                 blocktype, blocklen = struct.unpack(self.endian + "2I",
                                                     self.f.read(8))
+                print blocktype, blocklen
```
Comment (guedou) [inline scapy/utils.py]: Is it mandatory ?

Comment (p-l-) [inline scapy/utils.py]: forgotten debug statement :-/ Thanks!

**Pull request merged**